spec/support/request/shop_workflow.rb

```diff
@@ -1,13 +1,28 @@
 module ShopWorkflow
+  # If a spec uses `within` but we want to check something outside of that
+  # scope, we can search from the body element instead.
+  def find_body
+    page.all("body").first || page.find(:xpath, "ancestor::body")
+  end
+
   def wait_for_cart
-    first("#cart").click
-    within '.cart-sidebar' do
-      expect(page).to_not have_link "Updating cart..."
+    # Wait for debounce
+    #
+    # The auto-submit on these specific form elements (add to cart) now has a small built-in
+    # waiting period before submitting the data...
+    sleep 0.6
```

Comment:
I think this should be a method in itself as it was in the shopping_spec, that way the multiline comment becomes a method comment.

Reply:
Yes, I agree. I'll leave it in this case though because the mobile code will remove it again. It won't be needed.